B030 doesn't allow splatted exceptions

I *think* there's a false-report coming from B030 where we have some exceptions types defined in a constant. A minimal example:

```python
EXCEPTIONS_TO_IGNORE = (
    ValueError,
    TypeError,
)

try:
    pass
except (SyntaxError, *EXCEPTIONS_TO_IGNORE):
    pass
```

```console
$ flake8 t.py
t.py:8:1: B030 Except handlers should only be names of exception classes
```

This isn't incorrect, so I don't *think* the intention of B030 is to stop this kind of thing. 

<details>
<summary>versions etc</summary>

I'm using Macos 13.1, we're also seeing this in CI using Ubuntu 20.04.5.

```console
$ python --version --version
Python 3.10.8 (main, Oct 13 2022, 10:17:43) [Clang 14.0.0 (clang-1400.0.29.102)]

$ flake8 --version
6.0.0 (flake8-bugbear: 23.2.13, mccabe: 0.7.0, pycodestyle: 2.10.0, pyflakes: 3.0.1) CPython 3.10.8 on Darwin
```

</details>

It also doesn't allow function calls:

```python
def exceptions_to_ignore():
    return (ValueError, TypeError)

try:
    pass
except exceptions_to_ignore():
    pass
```

```console
$ flake8 t.py
t.py:6:1: B030 Except handlers should only be names of exception classes
```

Whether or not this is a good way to write code or not is an open question, but should it be disallowed?

--- a/README.rst
+++ b/README.rst
@@ -325,6 +325,11 @@ MIT
 Change Log
 ----------
 
+Unreleased
+~~~~~~~~~~
+
+* B030: Allow calls and starred expressions in except handlers.
+
 23.2.13
 ~~~~~~~~~
 

--- a/bugbear.py
+++ b/bugbear.py
@@ -333,17 +333,25 @@ class BugBearVisitor(ast.NodeVisitor):
         handlers = _flatten_excepthandler(node.type)
         good_handlers = []
         bad_handlers = []
+        ignored_handlers = []
         for handler in handlers:
             if isinstance(handler, (ast.Name, ast.Attribute)):
                 good_handlers.append(handler)
+            elif isinstance(handler, (ast.Call, ast.Starred)):
+                ignored_handlers.append(handler)
             else:
                 bad_handlers.append(handler)
         if bad_handlers:
             self.errors.append(B030(node.lineno, node.col_offset))
         names = [_to_name_str(e) for e in good_handlers]
-        if len(names) == 0 and not bad_handlers:
+        if len(names) == 0 and not bad_handlers and not ignored_handlers:
             self.errors.append(B029(node.lineno, node.col_offset))
-        elif len(names) == 1 and not bad_handlers and isinstance(node.type, ast.Tuple):
+        elif (
+            len(names) == 1
+            and not bad_handlers
+            and not ignored_handlers
+            and isinstance(node.type, ast.Tuple)
+        ):
             self.errors.append(B013(node.lineno, node.col_offset, vars=names))
         else:
             maybe_error = _check_redundant_excepthandlers(names, node)

--- a/tests/b030.py
+++ b/tests/b030.py
@@ -12,3 +12,18 @@ try:
     pass
 except (1, ValueError):  # error
     pass
+
+try:
+    pass
+except (ValueError, *(RuntimeError, TypeError)):  # ok
+    pass
+
+
+def what_to_catch():
+    return (ValueError, TypeError)
+
+
+try:
+    pass
+except what_to_catch():  # ok
+    pass

